java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageReader.java

```diff
@@ -18,48 +18,50 @@
 
 package org.apache.arrow.vector.ipc.message;
 
-
-import io.netty.buffer.ArrowBuf;
-import org.apache.arrow.flatbuf.Message;
-import org.apache.arrow.memory.BufferAllocator;
-
 import java.io.IOException;
 
 /**
- * Interface for reading a sequence of messages.
+ * Abstract base class for reading a sequence of messages.
  */
-public interface MessageReader {
+public abstract class MessageReader<T extends MessageReadHolder> {
```

Comment:
Should we keep this an interface?

Reply:
The point of making this an abstract class was to include the logic of the Arrow stream format that a message is prefixed by a length, then has an optional body determined by a field in the message.